File: Code/HVIT/HVIT_EX/HVIT_OOP_TEST/OOP/SoThuc/SoThuc.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace SoThuc
{
    class SoThuc
    {
        private double _GiaTri;
        public double GiaTri
        {
            get { return _GiaTri; }
            set
            {
                _GiaTri = value;
                KiemTraSoDuong();
            }
        }
        public bool LaSoDuong { get; set; }
        public void KiemTraSoDuong()
        {
            LaSoDuong = GiaTri > 0 ? true : false;
        }
        public SoThuc()
        {
            Console.Write("Nhap so thuc: ");
            GiaTri = double.Parse(Console.ReadLine());
        }

        public static SoThuc TimMax(SoThuc a, SoThuc b, SoThuc c)
        {
            return a.GiaTri >= b.GiaTri && a.GiaTri >= c.GiaTri ? a : b.GiaTri >= c.GiaTri && b.GiaTri >= a.GiaTri ? b : c;
        }

        public double TinhCanBacN(int n)
        {
            return Math.Pow(GiaTri, 1.0 / n);
        }
    }
}

File: Code/HVIT/HVIT_JAVA/src/JavaPractice/JavaOOPBasic/SoThuc/SoThuc.java
package JavaPractice.JavaOOPBasic.SoThuc;

public class SoThuc {
    private double GiaTri;
    private boolean LaSoDuong;

    public double getGiaTri() {
        return GiaTri;
    }

    public void setGiaTri(double giaTri) {
        GiaTri = giaTri;
        setLaSoDuong(GiaTri > 0 ? true : false);
    }

    public boolean isLaSoDuong() {
        return LaSoDuong;
    }

    public void setLaSoDuong(boolean laSoDuong) {
        LaSoDuong = laSoDuong;
    }

    public SoThuc(double gt) {
        setGiaTri(gt);
    }

    public static SoThuc TimMax(SoThuc a, SoThuc b, SoThuc c) {
        return (a.getGiaTri() > b.getGiaTri() && a.getGiaTri() > c.getGiaTri() ? a : b.getGiaTri() > a.getGiaTri() && b.getGiaTri() > c.getGiaTri() ? b : c);
    }

    public double TimCanBacN(int n) {
        return Math.pow(GiaTri, 1 * 1.0 / n);
    }
}
